src/util/request-manager.js

```diff
@@ -424,18 +425,27 @@ export default class RequestManager {
     const req = request(params);
     this.reporter.verbose(this.reporter.lang('verboseRequestStart', params.method, params.url));
 
-    req.on('error', onError);
-
     const queue = params.queue;
-    if (queue) {
-      req.on('data', queue.stillActive.bind(queue));
-    }
+
+    req.on('error', onError);
+    req.on('data', (data) => this.handleRequestData(data, queue));
 
     if (params.process) {
       params.process(req, resolve, reject);
     }
   }
 
+  handleRequestData(data: object, queue: object) {
+
+    if (data && data.length > 0) {
+      this.reporter.logNetworkUsage(data.length);
+    }
+
+    if (queue) {
+      queue.stillActive.bind(queue);
```

Comment:
What does this line do?

Reply:
From my memory, this checks to see if there are still requests resolving and keeps the spinners active in the console reporter.

[Current implementation](https://github.com/yarnpkg/yarn/blob/ae3cf9971d73350d6b52a0bfb3cc708e7a22d80b/src/util/request-manager.js#L441)

Reply:
It doesn't seem to be doing anything to me since `.bind()` simply returns a new function but doesn't call it. Since this is a "dangling" expression, I don't think the returned function is used anywhere and `.bind()` shouldn't have any side effects. Am I missing something here?

Reply:
> It doesn't seem to be doing anything to me since .bind() simply returns a new function but doesn't call it. Since this is a "dangling" expression, I don't think the returned function is used anywhere and .bind() shouldn't have any side effects. Am I missing something here?

Nope, not missing anything. You're correct, it seems to be a 'dangling' expression, and wasn't doing anything. Thanks for catching that.